### PR TITLE
[Trivial] More tests, code coverage and duplicate code cleanup

### PR DESCRIPTION
**Pull request by MarcusDenker**
- add some more tests to improve code coverage
- move #allocatePage up in the SoilIndex hierarchy 
- tag some methods <ignoreForCoverage>

### CODE REVIEW

File: src/Soil-Serializer-Tests/SoilPrimitiveSerializationTest.class.st [hunk 1]
```diff
@@ -225,7 +225,7 @@ SoilPrimitiveSerializationTest >> testSerializationDateAndTime [
 	self assert: serialized equals: #[29 5 10 0 0 5 0].
 	self assert: (serialized at: 1) equals: TypeCodeDateAndTime.
 
-	materialized := SoilBasicMaterializer materializeFromBytes: serialized.
+	materialized := SoilMaterializer materializeFromBytes: serialized.
```
Comment (noha) [inline src/Soil-Serializer-Tests/SoilPrimitiveSerializationTest.class.st:228]: The tests in this class are about the basic materializer. We have the basic materializer that can handle primitives while the full materializer can handle graphs. The basic materializer is used e.g. to read the database configuration which is plain
